barretenberg/cpp/src/barretenberg/ecc/fields/field_conversion.hpp

```diff
@@ -0,0 +1,289 @@
+{
+    return convert_from_bn254_frs(fr_vec, static_cast<T*>(nullptr));
+}
+
+/**
+ * @brief Conversion from transcript values to bb::frs
+ * @details We want to support the following types: bool, size_t, uint32_t, uint64_t, bb::fr, grumpkin::fr,
+ * curve::BN254::AffineElement, curve::Grumpkin::AffineElement, bb::Univariate<FF, N>, std::array<FF, N>, for
+ * FF = bb::fr/grumpkin::fr, and N is arbitrary.
+ * @tparam T
+ * @param val
+ * @return std::vector<bb::fr>
+ */
+template <std::integral T> std::vector<bb::fr> inline convert_to_bn254_frs(const T& val)
+{
+    std::vector<bb::fr> fr_vec{ val };
+    return fr_vec;
+}
+
+std::vector<bb::fr> inline convert_to_bn254_frs(const grumpkin::fr& val)
+{
+    auto fr_arr = convert_grumpkin_fr_to_bn254_frs(val);
+    std::vector<bb::fr> fr_vec(fr_arr.begin(), fr_arr.end());
+    return fr_vec;
+}
+
+std::vector<bb::fr> inline convert_to_bn254_frs(const bb::fr& val)
+{
+    std::vector<bb::fr> fr_vec{ val };
+    return fr_vec;
+}
+
+std::vector<bb::fr> inline convert_to_bn254_frs(const curve::BN254::AffineElement& val)
+{
+    auto fr_vec_x = convert_to_bn254_frs(val.x);
+    auto fr_vec_y = convert_to_bn254_frs(val.y);
+    std::vector<bb::fr> fr_vec(fr_vec_x.begin(), fr_vec_x.end());
+    fr_vec.insert(fr_vec.end(), fr_vec_y.begin(), fr_vec_y.end());
+    return fr_vec;
+}
+
+std::vector<bb::fr> inline convert_to_bn254_frs(const curve::Grumpkin::AffineElement& val)
+{
+    auto fr_vec_x = convert_to_bn254_frs(val.x);
+    auto fr_vec_y = convert_to_bn254_frs(val.y);
+    std::vector<bb::fr> fr_vec(fr_vec_x.begin(), fr_vec_x.end());
+    fr_vec.insert(fr_vec.end(), fr_vec_y.begin(), fr_vec_y.end());
+    return fr_vec;
+}
+
+template <size_t N> std::vector<bb::fr> inline convert_to_bn254_frs(const std::array<bb::fr, N>& val)
+{
+    std::vector<bb::fr> fr_vec(val.begin(), val.end());
+    return fr_vec;
+}
+
+template <size_t N> std::vector<bb::fr> inline convert_to_bn254_frs(const std::array<grumpkin::fr, N>& val)
+{
+    std::vector<bb::fr> fr_vec;
+    for (size_t i = 0; i < N; ++i) {
+        auto tmp_vec = convert_to_bn254_frs(val[i]);
+        fr_vec.insert(fr_vec.end(), tmp_vec.begin(), tmp_vec.end());
+    }
+    return fr_vec;
+}
+
+template <size_t N> std::vector<bb::fr> inline convert_to_bn254_frs(const bb::Univariate<bb::fr, N>& val)
+{
+    std::vector<bb::fr> fr_vec;
+    for (size_t i = 0; i < N; ++i) {
+        auto tmp_vec = convert_to_bn254_frs(val.evaluations[i]);
+        fr_vec.insert(fr_vec.end(), tmp_vec.begin(), tmp_vec.end());
+    }
+    return fr_vec;
+}
+
+template <size_t N> std::vector<bb::fr> inline convert_to_bn254_frs(const bb::Univariate<grumpkin::fr, N>& val)
+{
+    std::vector<bb::fr> fr_vec;
+    for (size_t i = 0; i < N; ++i) {
+        auto tmp_vec = convert_to_bn254_frs(val.evaluations[i]);
+        fr_vec.insert(fr_vec.end(), tmp_vec.begin(), tmp_vec.end());
+    }
+    return fr_vec;
+}
+
+// TODO: why is this needed here but not for the other 3 functions?
```

Comment:
seems like we have an asymmetry in our sumcheck prover/verifier. We send_to_verifier a ClaimedEvaluations (Flavor::AllValues) object, and receive_from_prover a std::array<FF, NUM_POLYNOMIALS> object. The underlying data of Flavor::AllValues ends up being a std::array<FF, NUM_POLYNOMIALS>, but its annoying that this asymmetry exists.